new resource `azurerm_elastic_san_volume_group` (#24166)
hashicorp/terraform-provider-azurerm: website/docs/r/elastic_san_volume_group.html.markdown

An `encryption` block supports the following arguments:

* `identity` - (Optional) An `identity` block as defined below.

* `key_vault_key_id` - (Optional) The Key Vault key URI for Customer Managed Key encryption, which can be either a full URI or a versionless URI.

---

An `identity` block supports the following arguments:

* `type` - (Required) Specifies the type of Managed Identity that should be assigned to this Elastic SAN Volume Group. Possible values are `SystemAssigned` and `UserAssigned`.

* `identity_ids` - (Optional) A list of the User Assigned Identity IDs that should be assigned to this Elastic SAN Volume Group.

---

A `network_rule` block supports the following arguments:

* `subnet_id` - (Required) The ID of the Subnet which should be allowed to access this Elastic SAN Volume Group.

* `action` - (Optional) The action to take when the Subnet attempts to access this Elastic SAN Volume Group. The only possible value is `Allow`. Defaults to `Allow`.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The ID of the Elastic SAN Volume Group.

---

An `encryption` block exports the following arguments:

> * `current_versioned_key_expiration_timestamp` - Current Versioned Key Expiration Timestamp of the Elastic SAN Volume Group.

Reviewer: ```suggestion
* `current_versioned_key_expiration_timestamp` - The timestamp of the expiration time for the current version of the customer managed key.
```

> * `current_versioned_key_id` - Current Versioned Key ID.

Reviewer: ```suggestion
* `current_versioned_key_id` - The ID of the current versioned Key Vault Key in use.
```

> * `last_key_rotation_timestamp` - Last Key Rotation Timestamp.

Reviewer: ```suggestion
* `last_key_rotation_timestamp` - The timestamp of the last rotation of the Key Vault Key.
```

---

An `identity` block exports the following arguments:

> * `principal_id` - The Principal ID for the System-Assigned Managed Identity assigned to this Elastic SAN Volume Group.

Reviewer: ```suggestion
* `principal_id` - The Principal ID associated with the Managed Service Identity assigned to this Elastic SAN Volume Group.
```

> * `tenant_id` - The Tenant ID for the System-Assigned Managed Identity assigned to this Elastic SAN Volume Group.

Reviewer: ```suggestion
* `tenant_id` - The Tenant ID associated with this Managed Service Identity assigned to this Elastic SAN Volume Group.
```

* `network_rule` - (Optional) One or more `network_rule` blocks as defined below.

> * `protocol_type` - (Optional) Specifies the type of the storage target. The only possible values are `Iscsi`. Defaults to `Iscsi`.

Reviewer: ```suggestion
* `protocol_type` - (Optional) Specifies the type of the storage target. The only possible value is `Iscsi`. Defaults to `Iscsi`.
```
